Student/StudentE2ETests/Offline/ContentSyncTests.swift

```diff
@@ -426,4 +426,123 @@ class OfflineContentSyncTests: OfflineE2ETest {
+        let isOffline = setNetworkStateOffline()
+        XCTAssertTrue(isOffline)
+
+        let offlineLineImage = Dashboard.offlineLine.waitUntil(.visible)
+        XCTAssertTrue(offlineLineImage.isVisible)
+
+        offlineCourseCard.hit()
+        assignmentsButton.waitUntil(.visible, timeout: 90)
+        XCTAssertTrue(assignmentsButton.isVisible)
+
+        // MARK: Navigate to Assignment List Page
+        assignmentsButton.hit()
+        let navBar = Assignment.navBar(course: course).waitUntil(.visible)
+        
+        XCTAssertTrue(navBar.isVisible)
+        let assignmentButton = Assignment.assignmentButton(assignment: assignment).waitUntil(.visible)
+        XCTAssertTrue(assignmentButton.isVisible)
+        XCTAssertTrue(assignmentButton.hasLabel(label: assignment.name + ", No Due Date"))
+        
+        // MARK: Tap on the assignment and check details
+        assignmentButton.hit()
+        let nameLabel = DetailsHelper.name.waitUntil(.visible)
+        XCTAssertTrue(nameLabel.isVisible)
+        XCTAssertTrue(nameLabel.hasLabel(label: assignment.name))
+
+        let pointsLabel = DetailsHelper.points.waitUntil(.visible)
+        XCTAssertTrue(pointsLabel.isVisible)
+        XCTAssertTrue(pointsLabel.hasLabel(label: "\(assignment.points_possible!) pt"))
+
+        let statusLabel = DetailsHelper.status.waitUntil(.visible)
+        XCTAssertTrue(statusLabel.isVisible)
+        XCTAssertTrue(statusLabel.hasLabel(label: "Not Submitted"))
+
+        let dueLabel = DetailsHelper.due.waitUntil(.visible)
+        XCTAssertTrue(dueLabel.isVisible)
+        XCTAssertTrue(dueLabel.hasLabel(label: "No Due Date"))
+
+        let submissionTypesLabel = DetailsHelper.submissionTypes.waitUntil(.visible)
+        XCTAssertTrue(submissionTypesLabel.isVisible)
+        XCTAssertTrue(submissionTypesLabel.hasLabel(label: "Text Entry"))
+
+        let descriptionLabel = DetailsHelper.description(assignment: assignment).waitUntil(.visible)
+        XCTAssertTrue(descriptionLabel.isVisible)
+        XCTAssertTrue(descriptionLabel.hasLabel(label: assignment.description!))
+
+        let submitAssignmentButton = DetailsHelper.submitAssignmentButton.waitUntil(.visible)
+        XCTAssertTrue(submitAssignmentButton.isVisible)
+        XCTAssertTrue(submitAssignmentButton.hasLabel(label: "Submit Assignment"))
+        let submissionButton = DetailsHelper.submissionAndRubricButton.waitUntil(.visible)
+        XCTAssertTrue(submissionButton.isVisible)
+        
+        
+        // MARK: Tap on the 'Submission & Rubric' and 'Submit Assignment' buttons and check if the offline mode dialog displayed and dismiss the dialog by hit the 'OK' button on it
+        submissionButton.hit()
+        OfflineHelper.assertOfflineAlertDialogContent()
+        OfflineHelper.offlineModeAlertOkButton.hit()
+        
+        submitAssignmentButton.hit()
+        OfflineHelper.assertOfflineAlertDialogContent()
+        OfflineHelper.offlineModeAlertOkButton.hit()
```

Comment:
There's no point in hitting the button if we don't assert to anything after it happens. So you can rather delete this line or assert to something after it got tapped (`Offline.offlineModeAlertOkButton.waitUntil(.vanished).isVanished`)